lemmy_apub/src/objects/mod.rs

```diff
@@ -36,3 +39,22 @@ where
     Err(anyhow!("Cant convert object to tombstone if it wasnt deleted").into())
   }
 }
+
+pub(in crate::objects) fn check_object_domain<T, Kind>(
+  apub: &T,
+  expected_domain: Option<Url>,
+) -> Result<String, LemmyError>
+where
+  T: Base + AsBase<Kind>,
+{
+  let actor_id = if let Some(url) = expected_domain {
+    check_is_apub_id_valid(&url)?;
```

Comment:
I think this was previously not checked at all in case someone searched for an object.

Reply:
I was wrong about this so I removed the line again. The check is actually done in `fetch_remote_object()`. Just shows again that this logic is too complicated.